assets/src/components/with-meta-block-edit.js

```diff
@@ -0,0 +1,178 @@
+		setTextColor,
+		toggleSelection,
+		tagName,
+	} = props;
+
+	const {
+		align,
+		ampFontFamily,
+		customFontSize,
+		height,
+		width,
+		opacity,
+	} = attributes;
+
+	const minTextHeight = 20;
+	const minTextWidth = 30;
+	const userFontSize = fontSize.size ? fontSize.size + 'px' : '2rem';
+
+	const ContentTag = tagName;
+
+	return (
+		<Fragment>
+			<BlockControls>
+				<AlignmentToolbar
+					value={ align }
+					onChange={ ( value ) => setAttributes( { align: value } ) }
+				/>
+			</BlockControls>
+			<InspectorControls>
+				<TextSettings
+					fontFamily={ ampFontFamily }
+					setFontFamily={ ( value ) => {
+						maybeEnqueueFontStyle( value );
+						setAttributes( { ampFontFamily: value } );
+					} }
+					fontSize={ fontSize }
+					setFontSize={ setFontSize }
+					customFontSize={ customFontSize }
+				/>
+				<ColorSettings
+					backgroundColor={ backgroundColor }
+					setBackgroundColor={ setBackgroundColor }
+					textColor={ textColor }
+					setTextColor={ setTextColor }
+					fontSize={ fontSize }
+					opacity={ opacity }
+					setOpacity={ ( value ) => setAttributes( { opacity: value } ) }
+				/>
+			</InspectorControls>
+			<ResizableBox
+				isSelected={ isSelected }
+				width={ width }
+				height={ height }
+				minHeight={ minTextHeight }
+				minWidth={ minTextWidth }
+				onResizeStop={ ( value ) => {
+					setAttributes( value );
+					toggleSelection( true );
+				} }
+				onResizeStart={ () => {
+					toggleSelection( false );
+				} }
+			>
+				<ContentTag
+					style={ {
+						backgroundColor: backgroundColor.color,
+						color: textColor.color,
+						fontSize: userFontSize,
+						textAlign: align,
+					} }
+					className={ classnames( className, {
+						'has-text-color': textColor.color,
+						'has-background': backgroundColor.color,
+						[ backgroundColor.class ]: backgroundColor.class,
+						[ textColor.class ]: textColor.class,
+						[ fontSize.class ]: fontSize.class,
+					} ) }
+				>
+					{ blockContent }
+				</ContentTag>
+			</ResizableBox>
+		</Fragment>
+	);
+};
+
+export default ( { attribute, tagName, isEditable } ) => {
+	return compose(
+		withSelect( ( select ) => {
+			const { getEditedPostAttribute } = select( 'core/editor' );
+			const { getAuthors } = select( 'core' );
+
+			const attributeValue = getEditedPostAttribute( attribute );
+
+			let blockContent;
+
+			// Todo: Maybe pass callbacks as props instead.
+			switch ( attribute ) {
+				case 'date':
+					// Disable reason: false positive because of the two leading underscores.
+					const dateFormat = __experimentalGetSettings().formats.date; // eslint-disable-line no-restricted-syntax
+					blockContent = dateI18n( dateFormat, attributeValue );
+
+					break;
+				case 'author':
+					const author = getAuthors().find( ( { id } ) => id === attributeValue );
+
+					blockContent = author ? author.name : __( 'Unknown', 'amp' );
+
+					break;
+				default:
+					blockContent = attributeValue;
```

Comment:
For the title, could we use the same placeholder as the title has?

Reply:
You mean `Add title`?

Reply:
Yes. Otherwise the block is empty and basically invisible when not selected.